### PR TITLE
feat: add new repository file handling

### PR DESCRIPTION
**Pull request by Morazzer**

### CODE REVIEW

File: src/main/java/dev/morazzer/cookiesmod/features/repository/files/RepositoryFileAccessor.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+
+public abstract class RepositoryFileAccessor {
+
+    protected static final Logger logger = LoggerFactory.getLogger("repository-files");
```
Comment (zFlxw) [inline src/main/java/dev/morazzer/cookiesmod/features/repository/files/RepositoryFileAccessor.java:14]: The naming should be in screaming snake case here.

File: src/main/java/dev/morazzer/cookiesmod/features/repository/files/RepositoryFileAccessor.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    /**
+     * Gets a directory as a list of {@linkplain com.google.gson.JsonElement}s.
+     *
+     * @param path The path to the directory (relative to the repository).
+     * @return The directory as list.
+     */
+    public abstract List<JsonElement> getDirectory(Path path);
+
+    /**
+     * Get a file as a {@linkplain com.google.gson.JsonElement}.
```
Comment (zFlxw) [inline src/main/java/dev/morazzer/cookiesmod/features/repository/files/RepositoryFileAccessor.java:49]:
```suggestion
     * Gets a file as a {@linkplain com.google.gson.JsonElement}.
```

File: src/main/java/dev/morazzer/cookiesmod/utils/StringUtils.java [hunk 3]
```diff
@@ -23,4 +23,18 @@ public static String escapeUnicode(String original) {
         return escaped.toString();
     }
 
+    /**
+     * Returns the byes as a hex string with lowercase characters.
```
Comment (zFlxw) [inline src/main/java/dev/morazzer/cookiesmod/utils/StringUtils.java:27]:
```suggestion
     * Returns the bytes as a hex string with lowercase characters.
```

File: src/main/java/dev/morazzer/cookiesmod/features/repository/items/recipe/RepositoryRecipeManager.java [hunk 4]
```diff
@@ -19,40 +16,30 @@
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.function.Predicate;
 import java.util.stream.Collectors;
-import java.util.stream.Stream;
 
 /**
  * Manger to handle all recipes in the repository.
```
Comment (zFlxw) [inline src/main/java/dev/morazzer/cookiesmod/features/repository/items/recipe/RepositoryRecipeManager.java:21]:
```suggestion
  * Manager to handle all recipes in the repository.
```

File: src/main/java/dev/morazzer/cookiesmod/features/repository/items/RepositoryItemManager.java [hunk 5]
```diff
@@ -82,43 +81,43 @@ public static void reloadItems() {
...
      */
     @SuppressWarnings("UnusedReturnValue")
-    public static boolean loadItem(Path path) {
-        return loadItem(path, (o) -> {});
+    public static boolean loadItem(JsonObject jsonObject) {
+        return loadItem(jsonObject, (o) -> {});
     }
 
     /**
      * Load one item from the local repository copy.
```
Comment (zFlxw) [inline src/main/java/dev/morazzer/cookiesmod/features/repository/items/RepositoryItemManager.java:108]:
```suggestion
      * Loads one item from the local repository copy.
```

**Review by zFlxw — CHANGES_REQUESTED** [review]
Some typos here and there, otherwise LGTM.

**Review by zFlxw — DISMISSED** [review]
LGTM

File: src/main/java/dev/morazzer/cookiesmod/features/repository/items/recipe/RepositoryRecipeManager.java [hunk 6]
```diff
@@ -19,40 +16,30 @@
...
+            if (!element.isJsonObject()) {
+                return;
+            }
+            JsonObject jsonObject = element.getAsJsonObject();
+            RecipeType type = RecipeType.valueOf(jsonObject.get("type").getAsString().toUpperCase());
+            RepositoryRecipe repositoryRecipe = type.getConstructor().create(jsonObject);
+
+            map.computeIfAbsent(repositoryRecipe.getOutput(), identifier -> new ArrayList<>()).add(repositoryRecipe);
         }
+        System.out.printf("Loaded %s recipes%n", map.size());
```
Comment (DreamTexX) [inline src/main/java/dev/morazzer/cookiesmod/features/repository/items/recipe/RepositoryRecipeManager.java:42]: Please use Logger here

**Review by DreamTexX — CHANGES_REQUESTED** [review]

**Review by DreamTexX — APPROVED** [review]
lgtm

**Pull request merged**